src/parser_json.c

```diff
@@ -1275,10 +1295,10 @@ lydjson_subtree_r(struct lyd_json_ctx *lydctx, struct lyd_node_inner *parent, st
     return ret;
 
 representation_error:
-    LOGVAL(ctx, LY_VLOG_LYD, parent, LYVE_SYNTAX_JSON,
-            "The %s \"%s\" is expected to be represented as JSON %s, but input data contains name/%s.",
+    LOG_LOCSET(ctx, NULL, (const struct lyd_node *)parent, NULL, NULL);
+    LOGVAL(ctx, LYVE_SYNTAX_JSON, "The %s \"%s\" is expected to be represented as JSON %s, but input data contains name/%s.",
             lys_nodetype2str(snode->nodetype), snode->name, expected, lyjson_token2str(status));
-
+    LOG_LOCBACK(ctx, 0, parent ? 1 : 0, 0, 0);
     ret = LY_EVALID;
     goto cleanup;
```

Comment:
LGTM seems not to like backwards `goto` so I would probably avoid it even though it is harmless in this case. Like this, for example:
```C
...
goto cleanup;

representation_error:
...
cleanup:
...
return ret;
```

Reply:
I'll try if LGTM will be fine with it, but I'm not sure that the code is more readable this way.